contracts/pre-propose/cwd-pre-propose-approval-single/src/contract.rs

```diff
@@ -0,0 +1,350 @@
+                contract_addr: proposal_module.into_string(),
+                msg: to_binary(&ProposeMessageInternal::Propose {
+                    title: proposal.msg.title,
+                    description: proposal.msg.description,
+                    msgs: proposal.msg.msgs,
+                    proposer: proposal.msg.proposer,
+                })?,
+                funds: vec![],
+            };
+
+            // Remove proposal
+            PENDING_PROPOSALS.remove(deps.storage, id);
+
+            Ok(Response::default()
+                .add_message(propose_messsage)
+                .add_attribute("method", "proposal_approved")
+                .add_attribute("proposal", id.to_string()))
+        }
+        None => Err(PreProposeError::ProposalNotFound {}),
+    }
+}
+
+pub fn execute_reject(
+    deps: DepsMut,
+    info: MessageInfo,
+    id: u64,
+) -> Result<Response, PreProposeError> {
+    // Check sender is the approver
+    let approver = APPROVER.load(deps.storage)?;
+    if approver != info.sender {
+        return Err(PreProposeError::Unauthorized {});
+    }
+
+    // Check proposal id exists
+    ensure!(
+        PENDING_PROPOSALS.has(deps.storage, id),
+        PreProposeError::ProposalNotFound {}
+    );
+
+    // Remove proposal
+    PENDING_PROPOSALS.remove(deps.storage, id);
+
+    // Handle deposit logic
+    match PrePropose::default().deposits.may_load(deps.storage, id)? {
+        Some((deposit_info, proposer)) => {
+            let messages = if let Some(ref deposit_info) = deposit_info {
+                // Refund can be issued if proposal if deposits are always refunded
+                // OnlyPassed and Never refund deposit policies do not apply here
+                if deposit_info.refund_policy == DepositRefundPolicy::Always {
+                    deposit_info.get_return_deposit_message(&proposer)?
+                } else {
+                    // If the proposer doesn't get the deposit, the DAO does.
+                    let dao = PrePropose::default().dao.load(deps.storage)?;
+                    deposit_info.get_return_deposit_message(&dao)?
+                }
+            } else {
+                // No deposit info for this proposal. Nothing to do.
+                vec![]
+            };
+
+            Ok(Response::default()
+                .add_attribute("method", "proposal_rejected")
+                .add_attribute("proposal", id.to_string())
+                .add_attribute("deposit_info", to_binary(&deposit_info)?.to_string())
+                .add_messages(messages))
+        }
+
+        // If we do not have a deposit for this proposal it was
+        // likely created before we were added to the proposal
+        // module. In that case, it's not our problem and we just
+        // do nothing.
+        None => Ok(Response::default()
+            .add_attribute("method", "proposal_rejected")
+            .add_attribute("proposal", id.to_string())),
+    }
+}
+
+pub fn execute_update_approver(
```

Comment:
nit: oak audit is going to say "pls do admin nomination flow"

Reply:
Lol.